rotkehlchen/exchanges/bitfinex.py

```diff
@@ -324,16 +328,31 @@ def _api_query_paginated(
             results.extend(cast('Iterable', results_))
             # NB: Copying the set before updating it prevents losing the call args values
             processed_result_ids = processed_result_ids.copy()
-            # type ignore is due to always having a trade link for bitfinex trades
-            processed_result_ids.update({int(result.link) for result in results_})  # type: ignore
+            newly_processed_results = set()
+            for result in results_:
+                if isinstance(result, AssetMovement):
+                    if result.unique_id is None:
+                        continue  # fees don't have a unique id
+
+                    newly_processed_results.add(int(result.unique_id))
```

Comment:
are those guaranteed to be integers? I'm not sure what casting to int does

Reply:
Do you need unique_id here? Isn't it enought with the event_identifier?

Reply:
> I'm not sure what casting to int does

These were being cast to int so that they would match here: https://github.com/rotki/rotki/blob/a282502d7ce3b365f4df8d3e1004cd38b190ff58/rotkehlchen/exchanges/bitfinex.py#L429 since the raw data is an int (at least in our test data).

> are those guaranteed to be integers?

In the test data they are integers but actually it looks like in the docs it says this (index 0 of the movement array) is a string: https://docs.bitfinex.com/reference/rest-auth-movements

I've changed this to handle these as strings, and instead of using a new unique_id property on the AssetMovement, I'm just putting it in extra_data. Not sure why I didn't think of doing it that way to begin with.

> Isn't it enought with the event_identifier?

Well, no that wouldn't really work here, since the `processed_result_ids` are being checked against the id in the raw data from the api to prevent processing the same event twice. The event_identifier is a hash of the location and the unique_id, so it can't be matched against anything in the raw data.